### DISCUSSION TITLE
Is it possible to use AWS key management service to decrypt the content encryption key ?

### DISCUSSION THREAD

**Discussion by tal-nagra** [panva/jose#491] [Q&A] [answered]
Hello,
I am currently attempting to use the `jose.jwtDecrypt` function.
Sample header:
`{"cty":"JWT","enc":"A256GCM","alg":"RSA-OAEP-256"}`
The CEK is encrypted using an RSA public key that originates from a key-pair that resides in the AWS key management service (RSAES_OAEP_SHA_256).
Therefore, to obtain the (decrypted) CEK the decryption needs to happen by invoking the AWS KMS sdk (by design, the private key never leaves the AWS key management service).

Is it possible to provide a custom function that performs the decryption of the CEK in place of the function below? 
https://github.com/panva/jose/blob/aff2f7c00f28b599ee72dd9f0a36c3783f1e195f/src/runtime/node/rsaes.ts#L66
Or achieve the desired CEK decryption via AWS in some other way ?  

Thank you in advance

**Comment by panva** [comment, marked as answer]
This is not in scope for this package. It only works with runtime's native key representations and crypto.

**Reply by tal-nagra** [reply]
I see. I guess I will take a look at alternative packages. Thanks for the swift response.
